packages/tooltips/src/styled/StyledLightTooltip.ts

```diff
@@ -5,27 +5,62 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import styled from 'styled-components';
-import classNames from 'classnames';
-import TooltipStyles from '@zendeskgarden/css-tooltips';
-import { retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import styled, { DefaultTheme, css, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import stripUnit from 'polished/lib/helpers/stripUnit';
 
 import { StyledTooltip, IStyledTooltipProps } from './StyledTooltip';
+import { StyledTitle } from './StyledTitle';
 
 const COMPONENT_ID = 'tooltip.light_tooltip';
 
+const arrowStyles = ({ theme, size }: IStyledTooltipProps & ThemeProps<DefaultTheme>) => {
+  let arrowFontSize = `${theme.space.base * 1.75 + 2}px`;
+
+  if (size === 'large') {
+    arrowFontSize = `${stripUnit(theme.fontSizes.sm) + 2}px`;
+  } else if (size === 'extra-large') {
+    arrowFontSize = `${theme.space.base * 4 + 2}px`;
+  }
+
+  return css`
+    &::before,
+    &::after {
+      font-size: ${arrowFontSize};
+    }
+  `;
+};
+
+const colorStyles = ({ theme }: ThemeProps<DefaultTheme>) => {
+  return css`
+    border: ${theme.borderWidths.sm} ${theme.borderStyles.solid}
+      ${getColor('neutralHue', 300, theme)};
+    box-shadow: 0 ${theme.space.base * 3}px ${theme.space.base * 5}px 0
+      ${getColor('chromeHue', 600, theme, 0.15)};
+    background-color: ${theme.colors.background};
+    color: ${getColor('neutralHue', 700, theme)};
+
+    ${StyledTitle} {
+      color: ${theme.colors.foreground};
+    }
+  `;
+};
+
 /**
  * Accepts all `<div>` props
  */
-export const StyledLightTooltip = styled(StyledTooltip).attrs(props => ({
+export const StyledLightTooltip = styled(StyledTooltip).attrs({
```

Comment:
From a styling POV, it would be better if all of this were incorporated into `StyledTooltip` – leveraging the `type` and removing `large`/`small` defaultProps – effectively removing several styling overrides. Keep the `styled` components agnostic with respect to size/color. Let the hi-abstraction component dictate which is expected, by default. (i.e. "light" tooltips shouldn't render at the "small" size and vice versa).

Reply:
I originally left the original CSS structure intact to reduce the number of changes in a single PR. I have now restructured this to be a single file that handles all permutations.